tools/shared.py

```diff
@@ -2781,6 +2781,108 @@ def run_binaryen_command(tool, infile, outfile=None, args=[], debug=False, stdou
   def run_wasm_opt(*args, **kwargs):
     return Building.run_binaryen_command('wasm-opt', *args, **kwargs)
 
+  @staticmethod
+  def do_wasm2c(infile):
```

Comment:
I don't like adding stuff to shared.py like this because:

(a) this is not shared.. it has exactly one call site
(b) shared.py get included absolutely everywhere.. we should try to keep it minimal.

I know that we are long way from (b) and we have history of throwing stuff in shared.py .. but I don't think thats good reason to make it worse.    I still like the idea of wasm2c.py which exactly one function in it.

Reply:
Yeah, I think I see what you mean. I moved it to a separate file now, let me know what you think.